### ISSUE TITLE
Memory Editor: Crashes using navigating through down arrow key with DataPreview option

### ISSUE THREAD

**Issue by JamesDelfini** [ocornut/imgui_club#26]
Using the [sample code given](https://github.com/ocornut/imgui_club/blob/master/imgui_memory_editor/imgui_memory_editor.h#L9-L14) to recreate the issue. Then set the values to below 15 both data and size with enabled option data preview, will cause the crash when key pressed down arrow.
```
// Sample 12
static char data[12];
size_t data_size = 12;
// and
mem_edit_1.OptShowDataPreview = true
```
This was due to the function of [DrawPreviewData()](https://github.com/ocornut/imgui_club/blob/master/imgui_memory_editor/imgui_memory_editor.h#L637), the reason was due to the **mem_size** is greater than the **addr**. The partial fix I did at the very first line of the function is to set the max memory size of the address(highlight selected):
`if (addr > mem_size) addr = mem_size;`

**Comment by JamesDelfini** [comment]
Correction: **addr** is greater than the **mem_size**, then set the **mem_size** as the maximum memory size which is 12 as the example.

Additional Information forgot to mention, the crash happens at this [line of code](https://github.com/ocornut/imgui_club/blob/master/imgui_memory_editor/imgui_memory_editor.h#L646), the size goes haywire which is the value of `4294967286`.

**Comment by ocornut** [comment]
Should be fixed by 94702c39b along with other issues.
Thanks for reporting!
